tools/site_admin/manage_special_days.php

```diff
@@ -33,16 +33,21 @@
 
         $spec_code = trim($_POST['spec_code']);
 
-        $source = new SpecialDay($spec_code);
+        if (strlen($spec_code) < 1) {
+            $errmsgs .= _("A value for Special Day ID is required. Please enter one.") . "<br>";
+        } elseif (preg_match("/[^A-Za-z0-9. -]/", $spec_code) === 1) {
+            $errmsgs .= _('The code must contain only alphanumeric characters . - or space.') . "<br>";
```

Comment:
Should the form field be a required field, with "Please enter a Special Day Code. Allowable characters are alphanumeric characters and the period, hyphen or space." as the error message if an attempt to create an empty special code is made, or one with bad characters? (I realize you just moved the existing check -- but this wouldn't be the first time we've updated a form to require fields.)